### DISCUSSION TITLE
How to use multiple machines for on-prem local installs?

### DISCUSSION THREAD

**Discussion by Abuhaliqa** [broadinstitute/seqr#2646] [Q&A] [answered]
Sorry if the question is beyond the team's scope here. In [option #2 for annotation in local install](https://github.com/broadinstitute/seqr/blob/master/deploy/LOCAL_INSTALL.md#option-2-annotate-and-load-on-prem), it mentions that it is possible to parallelize pipeline-runner. 
Since I am new to docker and Hail, I am at a loss at how to get 2 machines I am working on to share the workload to reduce the runtime for annotation. I would be grateful if someone could point me to the relevant documentation to archive just that.
Thanks,

**Comment by hanars** [comment]
Sorry if the documentation was unclear. The parallelization was referencing the fact that hail will use all available cores ("machines") so we recommend that if you choose to run the pipeline runner not on the cloud (which will almost always be a multi-cored instance) then we recommend using a high-powered multi-core machine locally. There is no way to manually control parallelization, it happens automatically

**Reply by Abuhaliqa** [reply]
Thank you very much for the clarification. Does this mean that there is no way to share the workload between 2 docker instances? 

If not, is there any way to use 2 machines beyond docker containers? 
I can see this version of [SeqR with Kubernetes](https://github.com/broadinstitute/seqr/tree/master/deploy/kubernetes) but can't find any documentation regarding that.

**Comment by hanars** [comment, marked as answer]
>  Does this mean that there is no way to share the workload between 2 docker instances?

Yes, that is not possible for seqr, or for docker in general

> If not, is there any way to use 2 machines beyond docker containers?

Yes, you can follow the instructions for [running the pipeline in dataproc](https://github.com/broadinstitute/seqr/blob/master/deploy/LOCAL_INSTALL.md#option-1-annotate-on-a-google-dataproc-cluster-then-load-in-to-an-on-prem-seqr-instance) . Dataproc can be configured to run with many workers, and our default configuration provides 14

> I can see this version of [SeqR with Kubernetes](https://github.com/broadinstitute/seqr/tree/master/deploy/kubernetes) but can't find any documentation regarding that.

Using kubernetes is not well supported for local seqr installations at this time. If you are interested in using the cloud to annotate I recommend you follow the instructions for using dataproc as above

**Reply by Abuhaliqa** [reply]
Thank you very much for the prompt reply.
